idx/cassandra/cassandra.go

```diff
@@ -170,12 +182,14 @@ func (c *CasIdx) InitBare() error {
 	return nil
 }
 
-// EnsureArchiveTableExists checks if the index archive table exists or not. If it does not exist and
-// the create-keyspace flag is true, then it will create it, if it doesn't exist and the create-keyspace
-// flag is false, then it will return an error. If the table exists then it just returns nil.
-// The index archive table is not required for Metrictank to run, it's only required by the
-// mt-index-prune utility to archive old metrics from the index.
-func (c *CasIdx) EnsureArchiveTableExists(session *gocql.Session) error {
+// EnsureTableExists checks if the specified table exists or not. If it does not exist and the
+// create-keyspace flag is true, then it will create it, if it doesn't exist and the create-keyspace
+// flag is false, then it will return an error. If the table exists then it just returns nil
+// sessions:    cassandra session
+// schemaFile:  file containing table definition
+// entryName:   identifier of the schema within the file
+// tableName:   name of the table in cassandra
+func (c *CasIdx) EnsureTableExists(session *gocql.Session, schemaFile, entryName, tableName string) error {
```

Comment:
instead of passing the schemaFile name, you should just pass an io.Reader that returns the schema file content.   This will make it easier to do testing and enable other callers to be able to call the method without having to first write a schemaFile to disk

Reply:
good idea, will do that

Reply:
Since we're using `go-toml` to read those schema files, I'd first need to read the schema file's correct entry into a string which then gets returned by `util.ReadEntry()`, wrap it into a reader, pass that reader into `EnsureTableExists`, and then read it back into a `string` again there. 
That seems like a bit of unnecessarily complication, wouldn't it be better to just pass a string with the table schema into `EnsureTableExists`? Then we still get the benefit of being able to unit test it, but we don't need to construct a reader and a buffer to read the reader into.

Reply:
What do you think about doing something like this:
https://github.com/grafana/metrictank/commit/0f0303ebd0feb5fdd44183d63b5c9ce4720fe9f9

I have pushed that into another branch, because I feel like it's getting out of scope of this PR. But with this we would have the advantages that initializing the cassandra index and stores becomes much more flexible because we can either provide a custom `io.Reader` to the `<config>.ParseSchemas()` functions, or we just directly set the schema we want in the according config struct.